### PR TITLE
Document the qjsc compiler part of QuickJS

### PR DESCRIPTION
**Pull request by jviotti**
As pointed-out by @guest271314 in
https://github.com/nodejs/single-executable/pull/8#issuecomment-1207225164.

Signed-off-by: Juan Cruz Viotti <jv@jviotti.com>

### CODE REVIEW

File: docs/existing-solutions.md [hunk 1]
```diff
@@ -23,3 +23,12 @@ PR!](https://github.com/nodejs/single-executable/edit/main/docs/existing-solutio
 [ASAR format]: https://github.com/electron/asar
 [`@zcbenz`]: https://github.com/zcbenz
 [monkey patching of `fs` in Electron to read from an ASAR]: https://github.com/electron/electron/blob/06a00b74e817a61f20e2734d50d8eb7bc9b099f6/lib/asar/fs-wrapper.ts
+
+Miscellaneous Related Tooling
+-----------------------------
+
+| Name           | Notes                                                |  
+|----------------|------------------------------------------------------|
+| [`qjsc`][qjsc] | A JavaScript to C byte-code compiler part of QuickJS |
```
Comment (RaisinTen) [inline docs/existing-solutions.md:32]: What do you mean by a JavaScript to C byte-code compiler? I haven't looked into the implementation but from my understanding, it is capable of compiling the JS file into an executable that also plays well with codesigining on macOS (haven't tried on Windows yet).

**Review by RaisinTen — COMMENTED** [review]
This needs to go before the "Did we miss any or got any details wrong?" part.

**Comment by jviotti** [comment]
Fair enough. Done!

File: docs/existing-solutions.md (hunk 1 above)
Comment (RaisinTen) [reply docs/existing-solutions.md:29]: I checked locally, it doesn't store the input js file as a string. It actually tokenized it and goes through a full compilation process I believe.

<details>

input js file:
```js
function f () {
  console.log('Hello, world!');
}

function g () {
  console.log('Holla, world!');
}

function main() {
  f();
  g();
}

main();
```

occurrence of `Hello, world` in the binary viewed through `xxd`:
```txt
000d74f0: f900 0000 0000 0000 0000 0000 0000 0000  ................
000d7500: 0208 0266 0267 086d 6169 6e0e 7465 7374  ...f.g.main.test
000d7510: 2e6a 730e 636f 6e73 6f6c 6506 6c6f 671a  .js.console.log.
000d7520: 4865 6c6c 6f2c 2077 6f72 6c64 211a 486f  Hello, world!.Ho
000d7530: 6c6c 612c 2077 6f72 6c64 210e 0006 00a0  lla, world!.....
000d7540: 0100 0100 0100 0332 01a2 0100 0000 3fe1  .......2......?.
```

</details>

Comment (jviotti) [reply docs/existing-solutions.md:29]: Ah, very interesting! I guess we can call it "a JavaScript to C compiler" then. Let me update the PR

Comment (RaisinTen) [reply docs/existing-solutions.md]: There's no C in the final output though, it's like a regular native executable.

File: docs/existing-solutions.md [hunk 2]
```diff
@@ -17,9 +17,18 @@ Virtual File System Implementations
...
 [ASAR format]: https://github.com/electron/asar
 [`@zcbenz`]: https://github.com/zcbenz
 [monkey patching of `fs` in Electron to read from an ASAR]: https://github.com/electron/electron/blob/06a00b74e817a61f20e2734d50d8eb7bc9b099f6/lib/asar/fs-wrapper.ts
+
+Miscellaneous Related Tooling
+-----------------------------
+
+| Name           | Notes                                      |  
+|----------------|--------------------------------------------|
+| [`qjsc`][qjsc] | A JavaScript to C compiler part of QuickJS |
```
Comment (RaisinTen) [inline docs/existing-solutions.md:29]:
```suggestion
| [`qjsc`][qjsc] | Compiles Javascript sources to executables with no external dependency. |
```
Copied the 5 point from https://bellard.org/quickjs/quickjs.html#Main-Features

Comment (jviotti) [reply docs/existing-solutions.md]: Makes sense!

**Review by RaisinTen — APPROVED** [review]

**Pull request merged**